pkg/kubeapiserver/default_storage_factory_builder.go

```diff
@@ -50,3 +54,56 @@ func NewStorageFactory(
 	resourceEncodingConfig = resourceconfig.MergeResourceEncodingConfigs(resourceEncodingConfig, resourceEncodingOverrides)
 	return serverstorage.NewDefaultStorageFactory(storageConfig, defaultMediaType, serializer, resourceEncodingConfig, apiResourceConfig, SpecialDefaultResourcePrefixes), nil
 }
+
+// MergeAPIResourceConfigs merges the given defaultAPIResourceConfig with the given resourceConfigOverrides.
+// Exclude the groups not registered in registry, and check if version is
+// not registered in group, then it will fail.
+func MergeAPIResourceConfigs(
```

Comment:
shouldn't we reject the case `len(tokens)==3` in `resourceconfig.MergeAPIResourceConfigs` and in the generic validation, and then filter those cases out here before calling `resourceconfig.MergeAPIResourceConfigs`?

Now you might pass a `group/v1/resource` to the generic options and they are ignored.

Reply:
Also I wonder whether this support for `group/v1/resource` shouldn't depend on each delegated API server, rather being allowed in general in the kube-apiserver chain. Do the aggregator and apiextensions API servers support resource-based disablement?

Reply:
>Do the aggregator and apiextensions API servers support resource-based disablement?

Good catch, now yes, but not my intention.

Reply:
>shouldn't we reject the case len(tokens)==3 in resourceconfig.MergeAPIResourceConfigs and in the generic validation, and then filter those cases out here before calling resourceconfig.MergeAPIResourceConfigs?

yes,  should do more validation.

Reply:
@sttts have one issue, if check len(tokens)==3 in generic validation, then it need duplicated code in kube-apiserver, e.g. parse group. Maybe I can put the check in `resourceconfig.MergeAPIResourceConfigs `

Reply:
was thinking about parsing the list and then splitting into len(token)==2 and those with ==3. Only pass down the former to the generic validation.